### ISSUE TITLE
How many images in your training dataset?

### ISSUE THREAD

**Issue by tongshiwen** [tancik/StegaStamp#3]
How many images in your training dataset?

**Comment by tancik** [comment]
The model was trained with a dataset of 100,000 images. Each training input also includes a random 100 bit string, so in practice the network will never see the exact same input.

I haven't tested it, but I imagine that it would be possible to train with a much smaller set of images (perhaps 10k) without impacting performance.

**Comment by tongshiwen** [comment]
Thank you very much! I've tested it on about 4,000 training images with  60 bit string, but it was not very good. I’ll add my dataset and try again. According to my experiment, I found than the simpler the training images, the higher the perceptual quality.My training dataset like that：
![zm_0_3](https://user-images.githubusercontent.com/28804340/57681136-b7426d80-7661-11e9-9c67-9e18abd0df6f.png)

**Comment by tancik** [comment]
The simpler training images may lead to higher perceptual quality, but I would also guess lower bit recovery accuracy (unless you have done a bit of hyperparameter tuning).

I would recommend using natural images for training. While training you should also keep your eye on the "bit_acc" value that is logged to tensorboard. It will start at .5 and should rise to > .8 within the first 5k iterations. If it remains at .5, the model is stuck in a local minimum and you will not be able to recover the hidden code. In this situation, the perceptual similarity will often continue to improve.

**Comment by 1165048017** [comment]
I trained the network with 5K images from mirflickr dataset. The decoding results were accurate. The training procedure took about 8 hours  with 140000 epochs.

**Comment by tongshiwen** [comment]
@tancik How to improve the perceptual quality? I did some training to improve the perceived quality, but it is not perfect. For example, the first is the original image and the second is the encoded image. After encoding, the perceived quality is very low.
![666_06](https://user-images.githubusercontent.com/28804340/58065259-af318300-7bb7-11e9-8d8f-69ae63c33b71.gif)
![666_06_hidden](https://user-images.githubusercontent.com/28804340/58065262-b22c7380-7bb7-11e9-9b1f-7a84b5fa843f.png)

**Comment by tancik** [comment]
The example image above is one of the hardest cases possible. The data must be hidden somewhere, any change a flat image (that is robust to printing) will be perceptually visible. Using natural images that have edges and textures will perform much better.

**Comment by zimenglan-sysu-512** [comment]
> The simpler training images may lead to higher perceptual quality, but I would also guess lower bit recovery accuracy (unless you have done a bit of hyperparameter tuning).
> 
> I would recommend using natural images for training. While training you should also keep your eye on the "bit_acc" value that is logged to tensorboard. It will start at .5 and should rise to > .8 within the first 5k iterations. If it remains at .5, the model is stuck in a local minimum and you will not be able to recover the hidden code. In this situation, the perceptual similarity will often continue to improve.

hi @tancik 
how to solve the case of "If it remains at .5, the model is stuck in a local minimum"? just change the learning rate?

**Comment by tancik** [comment]
I would start with learning rate, but you may need to adjust other hyperparameters. It is very case dependent.
